polyfill/lib/ecmascript.mjs

```diff
@@ -1423,14 +1423,26 @@ export const ES = ObjectAssign({}, ES2019, {
     return result;
   },
   ToBigInt: (arg) => {
-    if (bigInt.isInstance(arg)) return arg;
+    if (bigInt.isInstance(arg)) {
+      return arg;
+    }
+
     const prim = ES.ToPrimitive(arg, Number);
-    if (typeof prim === 'number') throw new TypeError('Use BigInt() to convert Number to BigInt');
-    try {
-      return bigInt(prim);
-    } catch (e) {
-      if (e instanceof Error && e.message.startsWith('Invalid integer')) throw new SyntaxError(e.message);
-      throw e;
+    switch (typeof prim) {
+      case 'undefined':
+      case 'object':
+      case 'number':
+      case 'symbol':
+        throw new TypeError(`cannot convert ${typeof arg} to bigint`);
+      case 'bigint':
+      case 'string':
+        return bigInt(prim);
```

Comment:
what's the expected result for the empty string?

Reply:
@ljharb I assumed here that `bigInt`'s string parser is compliant to the standard (so equivalent of https://tc39.es/ecma262/#sec-stringtobigint) but on second thought, that assumption might be incorrect, and now I'm questioning why we are using this polyfill in the first place? It has a lot of compliance issues...

Reply:
(as opposed to something like https://github.com/GoogleChromeLabs/jsbi)

Reply:
`bigInt(string)` seems to work almost like the spec, except that it accepts strings like `1e3` which `BigInt('1e3')` does not.

After a chat discussion between me and @ryzokuken, I think rather than changing bigint libraries at this point, we should wait for Safari 14 to become mainstream and then use no library at all, possibly adding a hack in the meantime to disallow string input like `'1e3'`.